core/embed/rust/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+#![cfg_attr(not(test), no_std)]
+#![feature(never_type)]
+#![feature(unsize)]
+#![feature(coerce_unsized)]
+#![feature(dispatch_from_dyn)]
+#![deny(clippy::all)]
+#![deny(unsafe_op_in_unsafe_fn)]
+#![allow(dead_code)]
+
+mod error;
+#[macro_use]
+mod micropython;
+mod trezorhal;
+mod util;
+
+#[cfg(not(test))]
+use core::panic::PanicInfo;
+#[cfg(not(test))]
+use cstr_core::CStr;
+
+#[cfg(not(test))]
+#[panic_handler]
+fn panic(_info: &PanicInfo) -> ! {
+    // Although it would be ideal to use the original error message, ignoring it
+    // lets us avoid the `fmt` machinery and its code size and is also important for
+    // security reasons, as we do not always controls the message contents. We
+    // should also avoid printing "panic" or "rust" on the user screen to avoid any
```

Comment:
I don't know, it's a cryptocurrency wallet and a electronic device. If the Rust code panics, the user definitely shouldn't, and it's most probably not because of "rust" as they know it :) So "rs" seems good enough to identify that it's a Rust-code panic, and avoid both confusions. INB4 "not stealing when not aggressive" screenshots on Reddit.